lacework/resource_lacework_alert_profile.go

```diff
@@ -0,0 +1,224 @@
+package lacework
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/lacework/go-sdk/api"
+	"github.com/pkg/errors"
+)
+
+func resourceLaceworkAlertProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLaceworkAlertProfileCreate,
+		Read:   resourceLaceworkAlertProfileRead,
+		Update: resourceLaceworkAlertProfileUpdate,
+		Delete: resourceLaceworkAlertProfileDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: importLaceworkAlertProfile,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The name of the alert profile",
+				ValidateDiagFunc: StringDoesNotHavePrefix("LW_"),
+			},
+			"extends": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of existing alert profile from which this profile extends",
+			},
+			"alert": {
+				Type:        schema.TypeSet,
+				MinItems:    1,
+				Required:    true,
+				Description: "The list of alert templates",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "The name that policies can use to refer to this template when generating alerts",
+							Required:    true,
+						},
+						"event_name": {
+							Type:        schema.TypeString,
+							Description: "The name of the resulting alert",
+							Required:    true,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Description: "The summary of the resulting alert",
+							Required:    true,
+						},
+						"subject": {
+							Type:        schema.TypeString,
+							Description: "A high-level observation of the resulting alert",
+							Required:    true,
+						},
+					},
+				},
+			},
+			"fields": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A field is a declaration of a field to be mapped in from an LQL query.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceLaceworkAlertProfileCreate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		alerts   []api.AlertTemplate
+	)
+	err := castSchemaSetToArrayOfAlertTemplate(d, "alert", &alerts)
+	if err != nil {
+		return err
+	}
+
+	alertProfile := api.NewAlertProfile(d.Get("name").(string),
+		d.Get("extends").(string),
+		alerts)
```

Comment:
Do we need any error checking here when using `Get` and subsequently casting to string?

Reply:
We don't, we use this in all our resources. The resource itself determines the type, we know it's always a string